src/backend/dx11/src/lib.rs

```diff
@@ -1358,31 +1436,70 @@ fn intersection(a: &Range<u64>, b: &Range<u64>) -> Option<Range<u64>> {
     }
 }
 
-// TODO: implement flush/invalidate for bound images as well
-impl Memory {
-    pub fn resolve<R: hal::range::RangeArg<u64>>(&self, range: &R) -> Range<u64> {
-        *range.start().unwrap_or(&0) .. *range.end().unwrap_or(&self.size)
+impl MemorySync {
+    fn flush(&self, context: &ComPtr<d3d11::ID3D11DeviceContext>) {
+        if let Some(range) = intersection(&self.sync_range, &self.bound_range) {
```

Comment:
should it be here in the first place if `sync_range` and `bound_range` don't intersect?